xds/src/main/java/io/grpc/xds/client/XdsClientImpl.java

```diff
@@ -667,20 +895,28 @@ void onData(ParsedResource<T> parsedResource, String version, long updateTime,
       }
     }
 
+    private String getTarget() {
+      ControlPlaneClient activeCpc = getActiveCpc(authority);
+      return (activeCpc != null)
+             ? activeCpc.getServerInfo().target()
+             : "unknown";
+    }
+
     void onAbsent(@Nullable ProcessingTracker processingTracker) {
       if (respTimer != null && respTimer.isPending()) {  // too early to conclude absence
         return;
       }
 
       // Ignore deletion of State of the World resources when this feature is on,
       // and the resource is reusable.
+      ControlPlaneClient activeCpc = getActiveCpc(authority);
       boolean ignoreResourceDeletionEnabled =
-          serverInfo != null && serverInfo.ignoreResourceDeletion();
+          activeCpc != null && activeCpc.getServerInfo().ignoreResourceDeletion();
       if (ignoreResourceDeletionEnabled && type.isFullStateOfTheWorld() && data != null) {
         if (!resourceDeletionIgnored) {
           logger.log(XdsLogLevel.FORCE_WARNING,
               "xds server {0}: ignoring deletion for resource type {1} name {2}}",
-              serverInfo.target(), type, resource);
+              getTarget(), type, resource);
```

Comment:
Instead of re-discovering what caused this, can we just pass ControlPlaneClient, ServerInfo, or target into `onAbsent()`? We have that information in the two places we call it.

Reply:
Changed it to get target from activeCpc.

Reply:
Now passing in serverInfo